eos-install-firefox

```diff
@@ -0,0 +1,358 @@
+#!/bin/bash -e
+
+# Downloads Firefox and unzips to the user's home directory,
+# with a desktop launcher
+
+TEMP_DIR=`mktemp -d`
+
+exit_with_error() {
+    echo $1
+    rm -rf ${TEMP_DIR}
+    exit 1
+}
+
+USERID=$(id -u)
+if [ "$USERID" == "0" ]; then
+    exit_with_error "Program cannot be run as superuser"
+fi
+
+DEB_ARCH=`dpkg --print-architecture`
+case ${DEB_ARCH} in
+    amd64) ;; # Only 64-bit Intel architectures are supported
+    *) exit_with_error "Unsupported architecture ${DEB_ARCH}"
+       ;;
+esac
+
+DEST_DIR=~/.local/share/eos-third-party-apps/firefox
+FLASH_DIR=${DEST_DIR}/flash
+PLUGINS_DIR=~/.mozilla/plugins
+DESKTOP_DIR=~/.local/share/applications
+WGET_COMMAND="wget -c --tries=5"
+
+FIREFOX_URL="https://download.mozilla.org/?product=firefox-latest&os=linux64"
+FIREFOX_LOCALIZED_URL=
+FIREFOX_FILENAME=firefox.tar.xz
+FIREFOX_LANG=
+
+JAVA_BUILD=8u112-b15
+JAVA_VERSION=`echo ${JAVA_BUILD} | cut -f 1 -d -`
+JAVA_FILENAME=jre-${JAVA_VERSION}-linux-x64.tar.gz
+JAVA_URL_FILENAME=http://download.oracle.com/otn-pub/java/jdk/${JAVA_BUILD}/${JAVA_FILENAME}
+
+# While the following would be a faster download of the current version,
+# previous versions are often removed from the server, so it is safest
+# for an automated script to use the stable version in the archive
+# http://fpdownload.macromedia.com/get/flashplayer/pdc/11.2.202.643/install_flash_player_11_linux.x86_64.tar.gz
+FLASH_VERSION=11.2.202.643
+FLASH_SHA256SUM="1a79f38f7153d436083b5a6b5d0f38e0c4cd345e83e7d050152452057f83350a"
+FLASH_FILENAME=fp_${FLASH_VERSION}_archive.zip
+FLASH_URL_FILENAME=http://download.macromedia.com/pub/flashplayer/installers/archive/${FLASH_FILENAME}
+
+ICON_FILENAME=firefox.png
+ICON_URL_FILENAME=https://raw.githubusercontent.com/endlessm/eos-customer-support/master/data/${ICON_FILENAME}
+
+get_firefox_lang() {
+    # List of supported Firefox languages, from https://download-installer.cdn.mozilla.net/pub/firefox/releases/latest/README.txt
+    FIREFOX_LANGS="ach af an ar as ast az be bg bn-BD bn-IN br bs ca cs cy da de dsb el en-GB en-US en-ZA eo es-AR es-CL es-ES es-MX et eu fa ff fi fr fy-NL ga-IE gd gl gu-IN he hi-IN hr hsb hu hy-AM id is it kk km kn ko lij lt lv mai mk ml mr ms nb-NO nl nn-NO or pa-IN pl pt-BR pt-PT rm ro ru si sk sl son sq sr sv-SE ta te th tr uk uz vi xh zh-CN zh-TW"
+
+    # Find the closest supported Firefox for the user's current language
+    locale=`echo ${LANG} | cut -f 1 -d . | sed s/_/-/`
+    lang=`echo ${locale} | cut -f 1 -d -`
+    for l in ${FIREFOX_LANGS}
+    do
+	if [ ${l} == ${locale} ]
+	then
+            FIREFOX_LANG=${l}
+            break
+	fi
+    done
+
+    if [[ -z ${FIREFOX_LANG} ]]; then
+	for l in ${FIREFOX_LANGS}
```

Comment:
Also, there seems to be a TABS & SPACES indentation mix-up happening here and in other points of this file which is messing up the visualization on github. Without entering into the eternal flamewar on tabs vs spaces :-), it would be nice to make the file consistent.